core/src/test/scala/integration/kafka/api/TransactionsExpirationTest.scala

```diff
@@ -129,9 +129,11 @@ class TransactionsExpirationTest extends KafkaServerTestHarness {
     producer.flush()
 
     // Ensure producer IDs are added.
-    val pState = producerState
-    assertEquals(1, pState.size)
-    val oldProducerId = pState(0).producerId
+    var pState : List[ProducerState] = null
+    TestUtils.waitUntilTrue(() => { pState = producerState; pState.nonEmpty}, "Producer IDs for topic1 did not propagate quickly")
+    assertEquals(1, pState.size, "Unexpected producer to topic1")
```

Comment:
The error message here seems to expect more than one producer?

Reply:
The message is only printed when the state assertion fails, and the waitUntilTrue has already asserted that it is > 0.
The only case that this could be true is if there are more than one producer, which the error message calls out.

Reply:
Ah ok -- that makes sense. I guess there's no crazy race where it would expire again now. :)